src/trace.rs

```diff
@@ -0,0 +1,9 @@
+use crate::rc::Rc;
+
+/// TODO: document me!
+pub trait Trace: Sized {
+    /// TODO: document me!
+    fn yield_owned_rcs<F>(&self, mark: F)
+    where
+        F: for<'a> FnMut(&'a mut Rc<Self>);
```

Comment:
I'm not sure if the higher ranked lifetime bound is correct here. Do we want to enforce that the lifetime of the yielded `&mut Rc<T>` is the same as `&self`? Removing the `for<'a>` would disallow non-static `T` that have `&mut Rc<T>` fields which would be a good thing.